### ISSUE TITLE
shoulda-matcher could not found test class name when work with rails 5.2.3 minitest.

### ISSUE THREAD

**Issue by zw963** [thoughtbot/shoulda-matchers#1229]
### Gemfile

```rb
group :test do
  gem 'shoulda-matchers', require: false
end
```
### test_helper.rb

```rb
Shoulda::Matchers.configure do |config|
    config.integrate do |with|
      with.test_framework :minitest_5
      with.library :active_model
    end
  end
```

### test file

```rb
require 'test_helper'

class ChannelTest < ActiveSupport::TestCase
  should validate_presence_of :username
end
```

## Error

```
The matcher attempted to set :username on the Class to "", but that (Shoulda::Matchers::ActiveModel::AllowValueMatcher::AttributeDoesNotExistError)
attribute does not exist
```

Following is backtrace.

```sh
╰─ $ 1  rails test test/models/channel_test.rb
Running via Spring preloader in process 8336
Traceback (most recent call last):
        50: from -e:1:in `<main>'
        49: from /home/zw963/others/.rvm/rubies/ruby-2.6.1/lib/ruby/site_ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        48: from /home/zw963/others/.rvm/rubies/ruby-2.6.1/lib/ruby/site_ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        47: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:285:in `load'
        46: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:257:in `load_dependency'
        45: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:285:in `block in load'
        44: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:285:in `load'
        43: from /home/zw963/Airhost/airhost_ror/bin/rails:4:in `<top (required)>'
        42: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:291:in `require'
        41: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:257:in `load_dependency'
        40: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:291:in `block in require'
        39: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/backports-3.15.0/lib/backports/std_lib.rb:9:in `require_with_backports'
        38: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/backports-3.15.0/lib/backports/std_lib.rb:9:in `require'
        37: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/railties-5.2.3/lib/rails/commands.rb:18:in `<top (required)>'
        36: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/railties-5.2.3/lib/rails/command.rb:46:in `invoke'
        35: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/railties-5.2.3/lib/rails/command/base.rb:65:in `perform'
        34: from /home/public/Dropbox/common/ruby/lib/thor.rb:387:in `dispatch'
        33: from /home/public/Dropbox/common/ruby/lib/thor/invocation.rb:126:in `invoke_command'
        32: from /home/public/Dropbox/common/ruby/lib/thor/command.rb:27:in `run'
        31: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/railties-5.2.3/lib/rails/commands/test/test_command.rb:33:in `perform'
        30: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/railties-5.2.3/lib/rails/test_unit/runner.rb:39:in `run'
        29: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/railties-5.2.3/lib/rails/test_unit/runner.rb:50:in `load_tests'
        28: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/railties-5.2.3/lib/rails/test_unit/runner.rb:50:in `each'
        27: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/railties-5.2.3/lib/rails/test_unit/runner.rb:50:in `block in load_tests'
        26: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:291:in `require'
        25: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:257:in `load_dependency'
        24: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:291:in `block in require'
        23: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/backports-3.15.0/lib/backports/std_lib.rb:9:in `require_with_backports'
        22: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/backports-3.15.0/lib/backports/std_lib.rb:9:in `require'
        21: from /home/zw963/Airhost/airhost_ror/test/models/channel_test.rb:41:in `<top (required)>'
        20: from /home/zw963/Airhost/airhost_ror/test/models/channel_test.rb:44:in `<class:ChannelTest>'
        19: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/rspec-expectations-3.8.2/lib/rspec/expectations/syntax.rb:45:in `should'
        18: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/rspec-expectations-3.8.2/lib/rspec/expectations/handler.rb:48:in `handle_matcher'
        17: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/rspec-expectations-3.8.2/lib/rspec/expectations/handler.rb:27:in `with_matcher'
        16: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/rspec-expectations-3.8.2/lib/rspec/expectations/handler.rb:50:in `block in handle_matcher'
        15: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/shoulda-matchers-4.1.0/lib/shoulda/matchers/active_model/validate_presence_of_matcher.rb:155:in `matches?'
        14: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/shoulda-matchers-4.1.0/lib/shoulda/matchers/active_model/validate_presence_of_matcher.rb:155:in `all?'
        13: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/shoulda-matchers-4.1.0/lib/shoulda/matchers/active_model/validate_presence_of_matcher.rb:156:in `block in matches?'
        12: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/shoulda-matchers-4.1.0/lib/shoulda/matchers/active_model/validate_presence_of_matcher.rb:233:in `disallows_original_or_typecast_value?'
        11: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/shoulda-matchers-4.1.0/lib/shoulda/matchers/active_model/validation_matcher.rb:98:in `disallows_value_of'
        10: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/shoulda-matchers-4.1.0/lib/shoulda/matchers/active_model/validation_matcher.rb:162:in `run_allow_or_disallow_matcher'
         9: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/shoulda-matchers-4.1.0/lib/shoulda/matchers/active_model/disallow_value_matcher.rb:32:in `matches?'
         8: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/shoulda-matchers-4.1.0/lib/shoulda/matchers/active_model/allow_value_matcher.rb:401:in `does_not_match?'
         7: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/shoulda-matchers-4.1.0/lib/shoulda/matchers/active_model/allow_value_matcher.rb:534:in `run'
         6: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/shoulda-matchers-4.1.0/lib/shoulda/matchers/active_model/allow_value_matcher.rb:534:in `public_send'
         5: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/shoulda-matchers-4.1.0/lib/shoulda/matchers/active_model/allow_value_matcher/attribute_setters_and_validators.rb:24:in `first_passing'
         4: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/shoulda-matchers-4.1.0/lib/shoulda/matchers/active_model/allow_value_matcher/attribute_setters_and_validators.rb:24:in `detect'
         3: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/shoulda-matchers-4.1.0/lib/shoulda/matchers/active_model/allow_value_matcher/attribute_setters_and_validators.rb:24:in `each'
         2: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/shoulda-matchers-4.1.0/lib/shoulda/matchers/active_model/allow_value_matcher/attribute_setters_and_validators.rb:38:in `matches?'
         1: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/shoulda-matchers-4.1.0/lib/shoulda/matchers/active_model/allow_value_matcher/attribute_setter.rb:78:in `set!'
/home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/shoulda-matchers-4.1.0/lib/shoulda/matchers/active_model/allow_value_matcher/attribute_setter.rb:214:in `attribute_does_not_exist!': The matcher attempted to set :username on the Class to "", but that (Shoulda::Matchers::ActiveModel::AllowValueMatcher::AttributeDoesNotExistError)
attribute does not exist.

```

**Comment by zw963** [comment]
I guess the reason cause this is because i add "rspec-rails" into Gemfile too, after i remove `rspec-rails`, i get following error:

```sh
 ╰─ $ rails test test/models/channel_test.rb

Symmetric Encryption config not found.
To generate a new config file and key files: symmetric-encryption --generate --app-name airhost_ror

/home/zw963/Airhost/airhost_ror/config/initializers/backtrace_silencers.rb:10: warning: already initialized constant Rails::BacktraceCleaner::APP_DIRS_PATTERN
/home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/railties-5.2.3/lib/rails/backtrace_cleaner.rb:7: warning: previous definition of APP_DIRS_PATTERN was here
I, [2019-06-26T21:12:56.155017 #3640]  INFO -- sentry: ** [Raven] Raven 2.9.0 configured not to capture errors: Not configured to send/capture in environment 'test'
Running via Spring preloader in process 3699
Traceback (most recent call last):
        30: from -e:1:in `<main>'
        29: from /home/zw963/others/.rvm/rubies/ruby-2.6.1/lib/ruby/site_ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        28: from /home/zw963/others/.rvm/rubies/ruby-2.6.1/lib/ruby/site_ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        27: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:285:in `load'
        26: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:257:in `load_dependency'
        25: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:285:in `block in load'
        24: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:285:in `load'
        23: from /home/zw963/Airhost/airhost_ror/bin/rails:4:in `<top (required)>'
        22: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:291:in `require'
        21: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:257:in `load_dependency'
        20: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:291:in `block in require'
        19: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/backports-3.15.0/lib/backports/std_lib.rb:9:in `require_with_backports'
        18: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/backports-3.15.0/lib/backports/std_lib.rb:9:in `require'
        17: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/railties-5.2.3/lib/rails/commands.rb:18:in `<top (required)>'
        16: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/railties-5.2.3/lib/rails/command.rb:46:in `invoke'
        15: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/railties-5.2.3/lib/rails/command/base.rb:65:in `perform'
        14: from /home/public/Dropbox/common/ruby/lib/thor.rb:387:in `dispatch'
        13: from /home/public/Dropbox/common/ruby/lib/thor/invocation.rb:126:in `invoke_command'
        12: from /home/public/Dropbox/common/ruby/lib/thor/command.rb:27:in `run'
        11: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/railties-5.2.3/lib/rails/commands/test/test_command.rb:33:in `perform'
        10: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/railties-5.2.3/lib/rails/test_unit/runner.rb:39:in `run'
         9: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/railties-5.2.3/lib/rails/test_unit/runner.rb:50:in `load_tests'
         8: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/railties-5.2.3/lib/rails/test_unit/runner.rb:50:in `each'
         7: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/railties-5.2.3/lib/rails/test_unit/runner.rb:50:in `block in load_tests'
         6: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:291:in `require'
         5: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:257:in `load_dependency'
         4: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:291:in `block in require'
         3: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/backports-3.15.0/lib/backports/std_lib.rb:9:in `require_with_backports'
         2: from /home/zw963/others/.rvm/gems/ruby-2.6.1@airhost_ror/gems/backports-3.15.0/lib/backports/std_lib.rb:9:in `require'
         1: from /home/zw963/Airhost/airhost_ror/test/models/channel_test.rb:41:in `<top (required)>'
/home/zw963/Airhost/airhost_ror/test/models/channel_test.rb:43:in `<class:ChannelTest>': undefined method `should' for ChannelTest:Class (NoMethodError)

```

**Comment by zw963** [comment]
after add `shoulda-context`, it worked now.

```sh
 ╰─ $ cat Gemfile.lock |grep shoulda
    shoulda-context (1.2.2)
    shoulda-matchers (4.1.0)
  shoulda-context
  shoulda-matchers
```

So, maybe we need change following README.md to correct this info.

![image](https://user-images.githubusercontent.com/549126/60190976-1fda5800-9866-11e9-905a-927e7cbbda56.png)

**Comment by mcmire** [comment]
I (finally) released Shoulda 4.0.0 and shoulda-context 2.0.0, so all Shoulda gems should be up to date with one another and with modern Rails versions, and this problem should be solved. But let me know if you see any more issues.
